pkg/steps/multi_stage.go

```diff
@@ -338,10 +341,14 @@ func (s *multiStageTestStep) runSteps(
 	return utilerrors.NewAggregate(errs)
 }
 
-func (s *multiStageTestStep) generatePods(steps []api.LiteralTestStep, env []coreapi.EnvVar) ([]coreapi.Pod, error) {
+func (s *multiStageTestStep) generatePods(steps []api.LiteralTestStep, env []coreapi.EnvVar,
+	prevErrs []error) ([]coreapi.Pod, error) {
 	var ret []coreapi.Pod
 	var errs []error
 	for _, step := range steps {
+		if s.allowSkipOnSuccess && step.OptionalOnSuccess && len(prevErrs) == 0 {
```

Comment:
Since you don't actually need to know the value of the previous errors, why don't we pass in `hasFailedSteps bool` or similar?

Reply:
It's possible, yeah. But in the run() function I'll need to assign hasFailedSteps more times or call len(errs) more times. When we pass the whole array we can only call (len) once in this common function. WDYT?

I'll also add the multi-step config as you suggested above.

Reply:
I would prefer to pass the thing we care about -- the boolean -- instead of letting the downstream callers have access to a list of data they do not need.

Reply:
OK. Pushed a change to pass only a bool flag.